src/common/ceph_context.cc

```diff
@@ -569,6 +578,17 @@ PerfCountersCollection *CephContext::get_perfcounters_collection()
   return _perf_counters_collection;
 }
 
+void CephContext::enable_perf_counter()
+{
+  ceph_spin_lock(&_cct_perf_lock);
```

Comment:
Hmm, I think all we really need to protect is the _cct_perf assignment, though, right?  The rest can happen outside of the lock?